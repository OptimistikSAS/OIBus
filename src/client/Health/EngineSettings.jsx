import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { FaEllipsisV, FaCog, FaRedoAlt } from 'react-icons/fa'
import { RiShutDownFill } from 'react-icons/ri'

import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap'
import { Modal, Button } from 'react-bootstrap'
import { useHistory } from 'react-router-dom'

const EngineSettings = ({ onRestart, onShutdown }) => {
  const [restartShow, setRestartShow] = useState(false)
  const handleRestartClose = () => setRestartShow(false)
  const handleRestartShow = () => setRestartShow(true)

  const [shutdownShow, setShutdownShow] = useState(false)
  const handleShutdownClose = () => setShutdownShow(false)
  const handleShutdownShow = () => setShutdownShow(true)

  const [dropdownOpen, setDropdownOpen] = useState(false)
  const toggle = () => setDropdownOpen((prevState) => !prevState)

  const history = useHistory()

  const handleGoToConnector = (pathname) => {
    history.push({ pathname })
  }

  return (
    <>
      <div className="icon-menu">
        <Dropdown
          isOpen={dropdownOpen}
          toggle={toggle}
          direction="left"
        >
          <DropdownToggle size="sm" className="icon-dropdown">
            <FaEllipsisV id="dropdown-toggle" className="icon-dropdown-ellipsis" />
          </DropdownToggle>

          <DropdownMenu>
            <DropdownItem className="icon-dropdown-item" onClick={handleRestartShow}>
              <FaRedoAlt id="icon-restart" className="icon-dropdown-item" />
              Restart
            </DropdownItem>
            <Modal show={restartShow} onHide={handleRestartClose}>
              <Modal.Body>Confirm restart?</Modal.Body>
              <Modal.Footer>
                <Button
                  id="restart-button"
                  variant="secondary"
                  onClick={onRestart}
                >
                  Confirm
                </Button>
                <Button
                  id="cancel-restart"
                  variant="primary"
                  onClick={handleRestartClose}
                >
                  Cancel
                </Button>
              </Modal.Footer>
            </Modal>

            <DropdownItem className="icon-dropdown-item" onClick={handleShutdownShow}>
              <RiShutDownFill id="icon-shutdown" className="icon-dropdown-item" />
              Shutdown
            </DropdownItem>
            <Modal show={shutdownShow} onHide={handleShutdownClose}>
              <Modal.Body>Confirm shutdown?</Modal.Body>
              <Modal.Footer>
                <Button
                  id="shutdown-button"
                  variant="secondary"
                  onClick={onShutdown}
                >
                  Confirm
                </Button>
                <Button
                  id="cancel-shutdown"
                  variant="primary"
                  onClick={handleShutdownClose}
                >
                  Cancel
                </Button>
              </Modal.Footer>
            </Modal>

            <DropdownItem className="icon-dropdown-item">
              <div
                id="icon-settings"
                aria-hidden="true"
                role="button"
                onClick={() => {
                  handleGoToConnector('/Engine/')
                }}
              >
                <FaCog className="icon-dropdown-item" />
                Settings
              </div>
            </DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </div>
    </>
  )
}

EngineSettings.propTypes = {
  onRestart: PropTypes.func.isRequired,
  onShutdown: PropTypes.func.isRequired,
}

export default EngineSettings
