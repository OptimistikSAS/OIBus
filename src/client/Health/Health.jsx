import React from 'react'
import { Label, Row, Breadcrumb, BreadcrumbItem } from 'reactstrap'
import { FaSync } from 'react-icons/fa'
import Table from '../components/table/Table.jsx'
import apis from '../services/apis'
import { AlertContext } from '../context/AlertContext.jsx'
import { ConfigContext } from '../context/configContext.jsx'
import Overview from './Overview.jsx'
import utils from '../helpers/utils'

const Health = () => {
  const [status, setStatus] = React.useState([])
  const { setAlert } = React.useContext(AlertContext)
  const { activeConfig } = React.useContext(ConfigContext)
  const config = utils.jsonCopy(activeConfig)
  utils.replaceValues(config, ['password', 'secretKey'], '******')
  const engineName = activeConfig?.engine.engineName ?? ''

  /**
   * Acquire the status
   * @returns {void}
   */
  const fetchStatus = () => {
    apis
      .getStatus()
      .then((response) => {
        setStatus(response)
      })
      .catch((error) => {
        console.error(error)
        setAlert({ text: error.message, type: 'danger' })
      })
  }

  /**
   * Fetch status after render
   * @returns {void}
   */
  React.useEffect(() => {
    fetchStatus()
  }, [])

  /**
   * Generate string value from on object.
   * @param {Object[]} data - The object
   * @return {string} - The string value
   */
  const generateStringValueFromObject = (data) => {
    let stringValue = ''
    Object.entries(data).forEach(([key, value]) => {
      if (key !== 'name') {
        stringValue += stringValue ? ` / ${key}: ${value}` : `${key}: ${value}`
      }
    })
    return stringValue
  }

  /**
   * Generate row entry for the status table.
   * @param {string} key - The key
   * @param {string} value - The value
   * @return {[{name: *, value: *}, {name: string, value: *}]} - The table row
   */
  const generateRowEntry = (key, value) => [
    {
      name: key,
      value: key,
    },
    {
      name: 'value',
      value,
    },
  ]

  const tableRows = []
  Object.keys(status).forEach((key) => {
    if (Array.isArray(status[key])) {
      status[key].forEach((entry) => {
        tableRows.push(generateRowEntry(entry.name, generateStringValueFromObject(entry)))
      })
    } else {
      tableRows.push(generateRowEntry(key, status[key]))
    }
  })

  return (
    <>
      <Breadcrumb tag="h5">
        <BreadcrumbItem active tag="span">
          Home
        </BreadcrumbItem>
      </Breadcrumb>
      <Row>
        <Label>
          <span>
            {`${engineName} overview`}
            &nbsp;
            <FaSync className="oi-icon" onClick={fetchStatus} />
          </span>
        </Label>
      </Row>
      <Overview />
      <Row>
        <Label>
          <span>
            {`${engineName} health status`}
            &nbsp;
            <FaSync className="oi-icon" onClick={fetchStatus} />
          </span>
        </Label>
      </Row>
      <Row>{tableRows && <Table headers={[]} rows={tableRows} />}</Row>
    </>
  )
}
export default Health
