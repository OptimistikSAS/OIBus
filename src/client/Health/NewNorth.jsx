import React, { useState } from 'react'
import { Button, Col, Form, Popover, PopoverBody, Row } from 'reactstrap'
import { nanoid } from 'nanoid'
import { ConfigContext } from '../context/configContext.jsx'
import validationNorth from '../North/Form/North.validation'
import { OIbSelect, OIbText } from '../components/OIbForm'

const NewNorth = () => {
  const { newConfig, dispatchNewConfig, apiList } = React.useContext(ConfigContext)
  const [name, setName] = React.useState('')
  const [api, setApi] = React.useState(apiList[0])

  const [openPopover, setOpenPopover] = useState(false)
  const applications = newConfig?.north?.applications ?? []

  const togglePopover = () => setOpenPopover(!openPopover)

  const addApplication = () => {
    if (!validationNorth.application.isValidName(name, applications?.map((application) => application.name) ?? [])) {
      togglePopover(false)
      dispatchNewConfig({
        type: 'addRow',
        name: 'north.applications',
        value: {
          id: nanoid(),
          name,
          api,
          enabled: false,
        },
      })
    }
  }

  const handleChange = (attributeName, value) => {
    switch (attributeName) {
      case 'name':
        setName(value)
        break
      case 'api':
      default:
        setApi(value)
        break
    }
  }

  return (
    <>
      <Button
        id="add-north"
        className="inline-button autosize oi-north-button"
        size="sm"
        outline
      >
        + North
      </Button>
      <Popover
        className="pop-container"
        trigger="legacy"
        placement="left"
        target="add-north"
        isOpen={openPopover}
        toggle={togglePopover}
      >
        <PopoverBody>
          <Form>
            <Row>
              <Col md="7">
                <OIbText
                  label="Application Name"
                  value={name}
                  name="name"
                  onChange={handleChange}
                  defaultValue=""
                  valid={() => validationNorth.application.isValidName(name, applications?.map((application) => application.name) ?? [])}
                />
              </Col>
              <Col md="5">
                <OIbSelect label="API" value={api} name="api" options={apiList} defaultValue={apiList[0]} onChange={handleChange} />
              </Col>
            </Row>
            <br />
            <Col md="2">
              <Button size="sm" id="icon-add" className="oi-add-button" color="primary" onClick={addApplication}>
                Add
              </Button>
            </Col>
          </Form>
        </PopoverBody>
      </Popover>
    </>

  )
}

export default NewNorth
