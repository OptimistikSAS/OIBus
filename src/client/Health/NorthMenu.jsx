import React, { useState } from 'react'
import { nanoid } from 'nanoid'
import PropTypes from 'prop-types'
import { FaEllipsisV } from 'react-icons/fa'
import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap'
import { useHistory } from 'react-router-dom'
import { ConfigContext } from '../context/configContext.jsx'
import ConfirmationModal from '../components/ConfirmationModal.jsx'

const NorthMenu = ({ application }) => {
  const { newConfig, dispatchNewConfig } = React.useContext(ConfigContext)
  const history = useHistory()
  const applications = newConfig?.north?.applications ?? []
  const [modal, setModal] = useState(false)
  const [dropdownOpen, setDropdownOpen] = useState(false)

  const handleGoToConnector = (pathname) => {
    history.push({ pathname })
  }
  const handleDuplicateNorth = () => {
    const newName = `${application.name} copy`
    const countCopies = applications.filter((e) => e.name.startsWith(newName)).length
    dispatchNewConfig({
      type: 'addRow',
      name: 'north.applications',
      value: {
        ...application,
        id: nanoid(),
        name: `${newName}${countCopies > 0 ? countCopies + 1 : ''}`,
        enabled: false,
      },
    })
  }
  const handleDeleteConnector = (name) => {
    dispatchNewConfig({ type: 'deleteRow', name })
  }

  const title = 'Delete'
  const body = `Are you sure you want to delete ${application.name}?`
  const onConfirm = () => {
    handleDeleteConnector(`north.applications.${applications.findIndex(
      (element) => element.id === application.id,
    )}`)
  }

  return (
    <>

      <Dropdown
        isOpen={dropdownOpen}
        toggle={() => setDropdownOpen((prevState) => !prevState)}
        direction="down"
      >
        <DropdownToggle size="sm" className="icon-dropdown">
          <FaEllipsisV id="dropdown-toggle" className="icon-dropdown-ellipsis" />
        </DropdownToggle>

        <DropdownMenu style={{ textAlign: 'center' }}>
          <DropdownItem className="dropdown-item">
            <div
              id="oi-settings"
              aria-hidden="true"
              role="button"
              onClick={() => {
                handleGoToConnector(`/north/${application.id}`)
              }}
            >
              Settings
            </div>
          </DropdownItem>

          <DropdownItem className="dropdown-item">
            <div
              id="oi-duplicate"
              aria-hidden="true"
              role="button"
              onClick={() => {
                handleDuplicateNorth(application.id)
              }}
            >
              Duplicate
            </div>
          </DropdownItem>

          <DropdownItem className="dropdown-item" onClick={() => setModal(true)}>
            <div id="oi-delete">Delete</div>
          </DropdownItem>
          <ConfirmationModal title={title} body={body} onConfirm={onConfirm} isOpen={modal} toggle={() => setModal(false)} />
        </DropdownMenu>
      </Dropdown>
    </>
  )
}

NorthMenu.propTypes = { application: PropTypes.object.isRequired }

export default NorthMenu
