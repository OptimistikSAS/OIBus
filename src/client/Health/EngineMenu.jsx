import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { FaEllipsisV, FaCog, FaRedoAlt, FaPlus } from 'react-icons/fa'
import { RiShutDownFill } from 'react-icons/ri'
import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap'
import { useHistory } from 'react-router-dom'
import ConfirmationModal from '../components/ConfirmationModal.jsx'
import NewSouth from './NewSouth.jsx'
import NewNorth from './NewNorth.jsx'

const EngineMenu = ({ onRestart, onShutdown }) => {
  const [southModal, setSouthModal] = useState(false)
  const [northModal, setNorthModal] = useState(false)
  const [restartShow, setRestartShow] = useState(false)
  const [shutdownShow, setShutdownShow] = useState(false)
  const [dropdownOpen, setDropdownOpen] = useState(false)
  const toggle = () => setDropdownOpen((prevState) => !prevState)

  const history = useHistory()

  const handleGoToConnector = (pathname) => {
    history.push({ pathname })
  }
  const titleRestart = 'Restart'
  const titleShutDown = 'Shutdown'
  const bodyRestart = 'Confirm restart?'
  const bodyShutDown = 'Confirm shutdown?'

  return (
    <>
      <div className="icon-menu">
        <Dropdown
          isOpen={dropdownOpen}
          toggle={toggle}
          direction="left"
        >
          <DropdownToggle size="sm" className="icon-dropdown">
            <FaEllipsisV id="dropdown-toggle" className="icon-dropdown-ellipsis" />
          </DropdownToggle>

          <DropdownMenu>
            <DropdownItem className="icon-dropdown-item" onClick={() => setSouthModal(true)}>
              <FaPlus id="icon-add-south" className="icon-dropdown-item" />
              Add South
            </DropdownItem>

            <DropdownItem className="icon-dropdown-item" onClick={() => setNorthModal(true)}>
              <FaPlus id="icon-add-north" className="icon-dropdown-item" />
              Add North
            </DropdownItem>

            <DropdownItem className="icon-dropdown-item">
              <div
                id="icon-settings"
                aria-hidden="true"
                role="button"
                onClick={() => {
                  handleGoToConnector('/Engine/')
                }}
              >
                <FaCog className="icon-dropdown-item" />
                Settings
              </div>
            </DropdownItem>

            <DropdownItem className="icon-dropdown-item" onClick={() => setRestartShow(true)}>
              <FaRedoAlt id="icon-restart" className="icon-dropdown-item" />
              Restart
            </DropdownItem>
            <ConfirmationModal
              title={titleRestart}
              body={bodyRestart}
              onConfirm={onRestart}
              isOpen={restartShow}
              toggle={() => setRestartShow(false)}
            />

            <DropdownItem className="icon-dropdown-item" onClick={() => setShutdownShow(true)}>
              <RiShutDownFill id="icon-shutdown" className="icon-dropdown-item" />
              Shutdown
            </DropdownItem>
            <ConfirmationModal
              title={titleShutDown}
              body={bodyShutDown}
              onConfirm={onShutdown}
              isOpen={shutdownShow}
              toggle={() => setShutdownShow(false)}
            />
          </DropdownMenu>
        </Dropdown>
        <NewSouth modal={southModal} toggle={() => setSouthModal(false)} />
        <NewNorth modal={northModal} toggle={() => setNorthModal(false)} />
      </div>
    </>
  )
}

EngineMenu.propTypes = {
  onRestart: PropTypes.func.isRequired,
  onShutdown: PropTypes.func.isRequired,
}

export default EngineMenu
