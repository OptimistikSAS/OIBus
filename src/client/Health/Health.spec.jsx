import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

// need BrowserRouter so Link component is not complaining
import { BrowserRouter } from 'react-router-dom'

import apis from '../services/apis'
import Health from './Health.jsx'

import activeConfig from '../../../tests/testConfig'

let container

beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

// sample status (object returned by Server to give various informations on the behavior)
const status = {
  Version: 'x.x.x',
  Architecture: 'x64',
  CurrentDirectory: '/Users/jfh/Work/OIBus/tests',
  'Node Version': 'v12.14.0',
  Executable: '/usr/local/bin/node',
  ConfigFile: '/Users/jfh/Work/OIBus/tests/oibus.json',
  Copyright: '(c) Copyright 2019 Optimistik, all rights reserved.',
}
// mock get Status
let resolve
apis.getStatus = () => (
  new Promise((_resolve) => {
    resolve = _resolve
  })
)

React.useContext = jest.fn().mockReturnValue({ activeConfig })

describe('Health', () => {
  test('display Health page based on config and status', async () => {
    act(() => {
      ReactDOM.render(<BrowserRouter><Health /></BrowserRouter>, container)
    })
    expect(container).toMatchSnapshot()
    // resolve the call requested by useffect
    await act(async () => {
      resolve(status)
    })
    expect(container).toMatchSnapshot()
  })
})
