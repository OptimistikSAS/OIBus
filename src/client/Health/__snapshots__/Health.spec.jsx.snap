// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Health display Health page based on config and status 1`] = `
<div>
  <h5
    aria-label="breadcrumb"
    class=""
  >
    <ol
      class="breadcrumb"
    >
      <span
        aria-current="page"
        class="active breadcrumb-item"
      >
        Home
      </span>
    </ol>
  </h5>
  <div
    class="row"
  >
    <label
      class=""
    >
      <span>
        OIBus overview
         
        <svg
          class="oi-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M440.65 12.57l4 82.77A247.16 247.16 0 0 0 255.83 8C134.73 8 33.91 94.92 12.29 209.82A12 12 0 0 0 24.09 224h49.05a12 12 0 0 0 11.67-9.26 175.91 175.91 0 0 1 317-56.94l-101.46-4.86a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12H500a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12h-47.37a12 12 0 0 0-11.98 12.57zM255.83 432a175.61 175.61 0 0 1-146-77.8l101.8 4.87a12 12 0 0 0 12.57-12v-47.4a12 12 0 0 0-12-12H12a12 12 0 0 0-12 12V500a12 12 0 0 0 12 12h47.35a12 12 0 0 0 12-12.6l-4.15-82.57A247.17 247.17 0 0 0 255.83 504c121.11 0 221.93-86.92 243.55-201.82a12 12 0 0 0-11.8-14.18h-49.05a12 12 0 0 0-11.67 9.26A175.86 175.86 0 0 1 255.83 432z"
          />
        </svg>
      </span>
    </label>
  </div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="tight text-success col"
      >
        <div
          class="oi-box d-flex align-items-center"
        >
          <a
            href="/north/c"
          >
            <div>
              c
            </div>
            <div>
              (Console)
            </div>
          </a>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box d-flex align-items-center"
        >
          <a
            href="/north/monoiconnect"
          >
            <div>
              monoiconnect
            </div>
            <div>
              (OIConnect)
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="tight col"
      >
        <a
          href="/engine"
        >
          <div
            class="oi-box text-success d-flex align-items-center"
          >
            Engine
          </div>
        </a>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/CSVServer"
          >
            <div>
              CSVServer
            </div>
            <div>
              (CSV)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/MQTTServer"
          >
            <div>
              MQTTServer
            </div>
            <div>
              (MQTT)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SimulationServer"
          >
            <div>
              SimulationServer
            </div>
            <div>
              (OPCUA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SimulationServerBis"
          >
            <div>
              SimulationServerBis
            </div>
            <div>
              (OPCUA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (1)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/PLC-35"
          >
            <div>
              PLC-35
            </div>
            <div>
              (Modbus)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (5)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/PLC-42"
          >
            <div>
              PLC-42
            </div>
            <div>
              (Modbus)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-success col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/FolderScanner"
          >
            <div>
              FolderScanner
            </div>
            <div>
              (FolderScanner)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-secondary btn-sm disabled"
            disabled=""
          >
            File
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SQLDbToFile"
          >
            <div>
              SQLDbToFile
            </div>
            <div>
              (SQLDbToFile)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-secondary btn-sm disabled"
            disabled=""
          >
            File
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/OPC-HDA"
          >
            <div>
              OPC-HDA
            </div>
            <div>
              (OPCHDA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (3)
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <label
      class=""
    >
      <span>
        OIBus health status
         
        <svg
          class="oi-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M440.65 12.57l4 82.77A247.16 247.16 0 0 0 255.83 8C134.73 8 33.91 94.92 12.29 209.82A12 12 0 0 0 24.09 224h49.05a12 12 0 0 0 11.67-9.26 175.91 175.91 0 0 1 317-56.94l-101.46-4.86a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12H500a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12h-47.37a12 12 0 0 0-11.98 12.57zM255.83 432a175.61 175.61 0 0 1-146-77.8l101.8 4.87a12 12 0 0 0 12.57-12v-47.4a12 12 0 0 0-12-12H12a12 12 0 0 0-12 12V500a12 12 0 0 0 12 12h47.35a12 12 0 0 0 12-12.6l-4.15-82.57A247.17 247.17 0 0 0 255.83 504c121.11 0 221.93-86.92 243.55-201.82a12 12 0 0 0-11.8-14.18h-49.05a12 12 0 0 0-11.67 9.26A175.86 175.86 0 0 1 255.83 432z"
          />
        </svg>
      </span>
    </label>
  </div>
  <div
    class="row"
  >
    <table
      class="table table-sm table-striped table-hover"
    >
      <thead>
        <tr />
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Health display Health page based on config and status 2`] = `
<div>
  <h5
    aria-label="breadcrumb"
    class=""
  >
    <ol
      class="breadcrumb"
    >
      <span
        aria-current="page"
        class="active breadcrumb-item"
      >
        Home
      </span>
    </ol>
  </h5>
  <div
    class="row"
  >
    <label
      class=""
    >
      <span>
        OIBus overview
         
        <svg
          class="oi-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M440.65 12.57l4 82.77A247.16 247.16 0 0 0 255.83 8C134.73 8 33.91 94.92 12.29 209.82A12 12 0 0 0 24.09 224h49.05a12 12 0 0 0 11.67-9.26 175.91 175.91 0 0 1 317-56.94l-101.46-4.86a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12H500a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12h-47.37a12 12 0 0 0-11.98 12.57zM255.83 432a175.61 175.61 0 0 1-146-77.8l101.8 4.87a12 12 0 0 0 12.57-12v-47.4a12 12 0 0 0-12-12H12a12 12 0 0 0-12 12V500a12 12 0 0 0 12 12h47.35a12 12 0 0 0 12-12.6l-4.15-82.57A247.17 247.17 0 0 0 255.83 504c121.11 0 221.93-86.92 243.55-201.82a12 12 0 0 0-11.8-14.18h-49.05a12 12 0 0 0-11.67 9.26A175.86 175.86 0 0 1 255.83 432z"
          />
        </svg>
      </span>
    </label>
  </div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="tight text-success col"
      >
        <div
          class="oi-box d-flex align-items-center"
        >
          <a
            href="/north/c"
          >
            <div>
              c
            </div>
            <div>
              (Console)
            </div>
          </a>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box d-flex align-items-center"
        >
          <a
            href="/north/monoiconnect"
          >
            <div>
              monoiconnect
            </div>
            <div>
              (OIConnect)
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="tight col"
      >
        <a
          href="/engine"
        >
          <div
            class="oi-box text-success d-flex align-items-center"
          >
            Engine
          </div>
        </a>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/CSVServer"
          >
            <div>
              CSVServer
            </div>
            <div>
              (CSV)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/MQTTServer"
          >
            <div>
              MQTTServer
            </div>
            <div>
              (MQTT)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SimulationServer"
          >
            <div>
              SimulationServer
            </div>
            <div>
              (OPCUA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SimulationServerBis"
          >
            <div>
              SimulationServerBis
            </div>
            <div>
              (OPCUA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (1)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/PLC-35"
          >
            <div>
              PLC-35
            </div>
            <div>
              (Modbus)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (5)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/PLC-42"
          >
            <div>
              PLC-42
            </div>
            <div>
              (Modbus)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-success col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/FolderScanner"
          >
            <div>
              FolderScanner
            </div>
            <div>
              (FolderScanner)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-secondary btn-sm disabled"
            disabled=""
          >
            File
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SQLDbToFile"
          >
            <div>
              SQLDbToFile
            </div>
            <div>
              (SQLDbToFile)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-secondary btn-sm disabled"
            disabled=""
          >
            File
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/OPC-HDA"
          >
            <div>
              OPC-HDA
            </div>
            <div>
              (OPCHDA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (3)
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <label
      class=""
    >
      <span>
        OIBus health status
         
        <svg
          class="oi-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M440.65 12.57l4 82.77A247.16 247.16 0 0 0 255.83 8C134.73 8 33.91 94.92 12.29 209.82A12 12 0 0 0 24.09 224h49.05a12 12 0 0 0 11.67-9.26 175.91 175.91 0 0 1 317-56.94l-101.46-4.86a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12H500a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12h-47.37a12 12 0 0 0-11.98 12.57zM255.83 432a175.61 175.61 0 0 1-146-77.8l101.8 4.87a12 12 0 0 0 12.57-12v-47.4a12 12 0 0 0-12-12H12a12 12 0 0 0-12 12V500a12 12 0 0 0 12 12h47.35a12 12 0 0 0 12-12.6l-4.15-82.57A247.17 247.17 0 0 0 255.83 504c121.11 0 221.93-86.92 243.55-201.82a12 12 0 0 0-11.8-14.18h-49.05a12 12 0 0 0-11.67 9.26A175.86 175.86 0 0 1 255.83 432z"
          />
        </svg>
      </span>
    </label>
  </div>
  <div
    class="row"
  >
    <table
      class="table table-sm table-striped table-hover"
    >
      <thead>
        <tr />
      </thead>
      <tbody>
        <tr>
          <td>
            Version
          </td>
          <td>
            x.x.x
          </td>
          <td />
        </tr>
        <tr>
          <td>
            Architecture
          </td>
          <td>
            x64
          </td>
          <td />
        </tr>
        <tr>
          <td>
            CurrentDirectory
          </td>
          <td>
            /Users/jfh/Work/OIBus/tests
          </td>
          <td />
        </tr>
        <tr>
          <td>
            Node Version
          </td>
          <td>
            v12.14.0
          </td>
          <td />
        </tr>
        <tr>
          <td>
            Executable
          </td>
          <td>
            /usr/local/bin/node
          </td>
          <td />
        </tr>
        <tr>
          <td>
            ConfigFile
          </td>
          <td>
            /Users/jfh/Work/OIBus/tests/oibus.json
          </td>
          <td />
        </tr>
        <tr>
          <td>
            Copyright
          </td>
          <td>
            (c) Copyright 2019 Optimistik, all rights reserved.
          </td>
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
